### PR TITLE
airbyte-ci: fix format cli usage in airbyte-enterprise

### PR DESCRIPTION
**Pull request by postamar**
## What
Fixes bugs which prevented `airbyte-ci format fix all` from working in airbyte-enterprise when run straight from the CLI.
Also includes a QoL fix which makes maven run in batch mode instead of interactive mode, this makes the execution logs nicer.

## How
Inclusion and exclusion rules have been corrected to handle symlinks propertly.

## Review guide
One commit at a time.

## User Impact
None.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/consts.py [hunk 1]
```diff
@@ -62,7 +67,7 @@ class Formatter(Enum):
 WARM_UP_INCLUSIONS = {
     Formatter.JAVA: [
         "spotless-maven-pom.xml",
-        "tools/gradle/codestyle/java-google-style.xml",
+        "tools/",  # Whole directory instead of just 'java-google-style.xml' to support airbyte-enterprise.
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/consts.py:70]: I'm not sure why this change is required. And I think it could impact performance as we mount more things during the "warm up" phase.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py [hunk 2]
```diff
@@ -69,26 +69,32 @@ def build_container(
     return container
 
 
+def warmup_directory(dagger_client: dagger.Client, formatter: Formatter) -> Union[dagger.Directory, None]:
+    """Creates a Dagger directory which includes the warmup files for the given Formatter."""
+    include = WARM_UP_INCLUSIONS.get(formatter)
+    if not include:
+        return None
+    include = include + [f"{AIRBYTE_SUBMODULE_DIR_NAME}/{i}" for i in include]
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py:77]: Shouldn't we check thaht `AIRBYTE_SUBMODULE_DIR_NAME` exists first? I'm not sure if dagger fails if you tell it to include not existing path (as the submodule does not exists in the main repo)

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/consts.py (hunk 1 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/consts.py:70]: In airbyte-enterprise we have a symlink to `tools` and including a file inside of that only works if the symlink works.

I understand why you'd be worried by `tools` only contains 1.2 MB of files which basically never change. I think it's OK.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py (hunk 2 above)
Comment (postamar) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py:77]: No, conveniently enough, it's fine if the path doesn't exist!

**Review by alafanechere — APPROVED** [review]
Thanks for your replies! `format` ran successfully on this branch so it means you did not break it 😄
